sdk/src/test/tutorial/java/com/finbourne/lusid/tutorials/ibor/ReferencePortfolio.java

```diff
@@ -0,0 +1,149 @@
+        referencePortfoliosApi.createReferencePortfolio(TestDataUtilities.TutorialScope, createRequest);
+
+        List<ReferencePortfolioConstituentRequest> constituentRequests = new ArrayList<>();
+
+        // Prepare individual constituent requests
+        for (String instrumentId : instrumentWeights.keySet()) {
+            TreeMap<String, String> instrumentIdentifier = new TreeMap<>();
+            instrumentIdentifier.put(TestDataUtilities.LUSID_INSTRUMENT_IDENTIFIER, instrumentId);
+
+            constituentRequests.add(new ReferencePortfolioConstituentRequest()
+                    .instrumentIdentifiers(instrumentIdentifier)
+                    .weight(instrumentWeights.get(instrumentId))
+                    .currency("GBP")
+            );
+        }
+
+        // Prepare bulk constituent upsert request
+        UpsertReferencePortfolioConstituentsRequest upsertRequest = new UpsertReferencePortfolioConstituentsRequest()
+                .constituents(constituentRequests)
+                .weightType(UpsertReferencePortfolioConstituentsRequest.WeightTypeEnum.STATIC)
+                .effectiveFrom(effectiveDate.toString());
+
+        // Make the request
+        referencePortfoliosApi.upsertReferencePortfolioConstituents(TestDataUtilities.TutorialScope,
+                F40PortfolioCode,
+                upsertRequest);
+
+        // Get constituents of portfolio
+        List<ReferencePortfolioConstituent> constituents = referencePortfoliosApi.getReferencePortfolioConstituents(TestDataUtilities.TutorialScope,
+                F40PortfolioCode,
+                null,
+                null,
+                null).getConstituents();
+
+        // Sorting constituents by their LUIDs so we can compare to the request LUIDs
+        constituents.sort(Comparator.comparing(ReferencePortfolioConstituent::getInstrumentUid));
+
+        // Validate number of constituents matches what was inserted
+        assertEquals(constituents.size(), instrumentIds.size());
+
+        // Validate constituents match inserted instruments
+        Map<String, String> constituentInstrumentLuidPairs = IntStream.range(0, constituents.size())
+                .collect(TreeMap::new,
+                        (map, i) -> map.put(constituents.get(i).getInstrumentUid(), instrumentIds.get(i)),
+                        Map::putAll);
+
+        for (String constituent : constituentInstrumentLuidPairs.keySet())
+            assertEquals(constituent, constituentInstrumentLuidPairs.get(constituent));
+
+        // Validate that weights of the constituents match weights that were inserted
+        Map<Double, Double> constituentWeightPairs = IntStream.range(0, constituents.size())
+                .collect(TreeMap::new,
+                        (map, i) -> map.put(constituents.get(i).getWeight(), instrumentWeights.get(instrumentIds.get(i))),
+                        Map::putAll);
+
+        for (Double weight : constituentWeightPairs.keySet())
```

Comment:
I like that this compact form of having it in a for loop, but because this is public code and readability is of most importance, it maybe might be better to assert each element from the weights list individually?